src/olmo_core/model_ladder2/wsds_chinchilla_run_configurator.py

```diff
@@ -0,0 +1,215 @@
+import dataclasses
+import math
+from dataclasses import dataclass
+
+from olmo_core.aliases import PathOrStr
+from olmo_core.exceptions import OLMoConfigurationError
+from olmo_core.optim import (
+    WSDS,
+    OptimGroupOverride,
+    Scheduler,
+    SchedulerUnits,
+    SkipStepAdamWConfig,
+)
+from olmo_core.train import Duration
+
+from .base import RunConfigurator
+from .utils import format_count, format_tokens
+
+
+@dataclass(kw_only=True)
+class WSDSChinchillaRunConfigurator(RunConfigurator):
+    """
+    A run configurator that uses WSD-S learning rate scheduling and Chinchilla scaling laws.
+    """
+
```

Comment:
nit: the underlying chinchilla heuristic of 20 tok/param has some caveats, like it only applies to models trained with AdamW and is dataset dependent (was determined on the Pile iirc). Might be worth adding a disclaimer here so people dont assume that 20 tok/param is actually optimal for their dataset.